src/plugins/data/public/ui/query_string_input/language_switcher.tsx

```diff
@@ -52,15 +63,21 @@ export function QueryLanguageSwitcher(props: Props) {
       className="euiFormControlLayout__append kqlQueryBar__languageSwitcherButton"
       data-test-subj={'switchQueryLanguageButton'}
     >
-      {props.language === 'lucene' ? luceneLabel : kqlLabel}
+      {language === 'kuery' ? (
+        kqlLabel
+      ) : nonKqlMode === 'lucene' ? (
+        luceneLabel
+      ) : (
+        <EuiIcon type={'boxesVertical'} />
```

Comment:
Why wouldn't you use `nonKqlModeHelpText` in this case? 
If it wouldn't work for your case, then would it be useful to have a configurable `nonKqlLabel` prop?

Reply:
nonKqlModeHelpText actually goes into the popover, like "Uptime uses ..."
for the text mode idea is to add a non invasive icon instead of text.

![image](https://user-images.githubusercontent.com/3505601/108535663-dd3b7e00-72db-11eb-84e2-e17f04ae86ec.png)

Reply:
if some use cases arises where someone needs explicit text for Text mode, i guess at that point it can be added.